controllers/backup.go

```diff
@@ -232,34 +232,76 @@ func (r *ClusterGroupUpgradeReconciler) backupActive(ctx context.Context, cluste
 
 }
 
-// setBackupRequired sets conditions of backup required
-func (r *ClusterGroupUpgradeReconciler) setBackupRequired(clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade) {
-
-	meta.SetStatusCondition(
-		&clusterGroupUpgrade.Status.Conditions, metav1.Condition{
-			Type:    BackupStateDone,
-			Status:  metav1.ConditionFalse,
-			Reason:  "BackupNotDone",
-			Message: "Backup is required and not done"})
+// setBackupStartedCondition sets conditions of backup required
+func (r *ClusterGroupUpgradeReconciler) setBackupStartedCondition(clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade) {
+	utils.SetStatusCondition(
+		&clusterGroupUpgrade.Status.Conditions,
+		utils.ConditionTypes.BackupSuceeded,
+		utils.ConditionReasons.InProgress,
+		metav1.ConditionFalse,
+		"Backup is required and not done",
+	)
 }
 
 // checkAllBackupDone handles alleviation of BackupDone==False condition
 func (r *ClusterGroupUpgradeReconciler) checkAllBackupDone(
 	clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade) {
-	// Handle completion
-	if func() bool {
-		for _, state := range clusterGroupUpgrade.Status.Backup.Status {
-			if state != BackupStateSucceeded {
-				return false
-			}
+
+	// Counts for the various cluster states
+	var failedBackupCount int = 0
+	var progressingBackupCount int = 0
+	var successfulBackupCount int = 0
+
+	// Loop over all the clusters and take count of all their states
+	for _, state := range clusterGroupUpgrade.Status.Backup.Status {
+		switch state {
+		case BackupStateDone:
+			successfulBackupCount++
+		case BackupStateActive, BackupStateStarting, BackupStatePreparingToStart:
+			progressingBackupCount++
+		default:
+			failedBackupCount++
 		}
-		return true
-	}() {
-		meta.SetStatusCondition(
-			&clusterGroupUpgrade.Status.Conditions, metav1.Condition{
-				Type:    BackupStateDone,
-				Status:  metav1.ConditionTrue,
-				Reason:  "BackupCompleted",
-				Message: "Backup is completed"})
 	}
+
+	// Compare the total number of clusters to their status
+	switch len(clusterGroupUpgrade.Status.Backup.Status) {
+	// All clusters were successful
+	case successfulBackupCount:
+		utils.SetStatusCondition(
+			&clusterGroupUpgrade.Status.Conditions,
+			utils.ConditionTypes.BackupSuceeded,
+			utils.ConditionReasons.Completed,
+			metav1.ConditionTrue,
+			"Backup is completed for all clusters",
+		)
+	// All clusters failed
+	case failedBackupCount:
+		utils.SetStatusCondition(
+			&clusterGroupUpgrade.Status.Conditions,
+			utils.ConditionTypes.BackupSuceeded,
+			utils.ConditionReasons.Failed,
+			metav1.ConditionFalse,
+			"Backup failed for all clusters",
+		)
+	// All clusters are completed but some failed
+	case (failedBackupCount + successfulBackupCount):
+		utils.SetStatusCondition(
+			&clusterGroupUpgrade.Status.Conditions,
+			utils.ConditionTypes.BackupSuceeded,
+			utils.ConditionReasons.PartiallyDone,
+			metav1.ConditionTrue,
+			fmt.Sprintf("Backup failed for %d clusters", failedBackupCount),
```

Comment:
As i have mentioned before, we don't need to report the list of clusters that failed, since cluster specific status will highlight that, here we could just indicate some cluster failed

Reply:
Its not a list, its just the count. I don't see why we wouldn't include this when we have the information since its just a number and is somewhat informative. If I check the CGU and it says in progress for 1734 clusters then I know it'll be awhile but if it says in progress for 1 then I know its almost done

Reply:
I checked quickly and thought it was returning a list, count is fine. Ignore the other messages related to this.

Reply:
It was a list in the original PR however I changed it to count when I redid how I was accounting for the in progress clusters